### PR TITLE
Added paulpcb keyboard

### PR DESCRIPTION
**Pull request by AlfieLockrey**
## Description

Added the firmware for a newly created and sold PCB that offers encoder support in a reverse Alice layout and specific mounting viable for a 3D printed case.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/paulpcb/info.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
+{
+    "keyboard_name": "AWL PaulPCB",
+    "url": "https://github.com/AlfieLockrey",
+    "maintainer": "AlfieLockrey",
+    "width": 18.5,
+    "height": 6.6,
+    "layouts": {
+        "LAYOUT_paulpcb_default": {
```
Comment (zvecr) [inline keyboards/paulpcb/info.json:8]:
```suggestion
        "LAYOUT": {
```

File: keyboards/paulpcb/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_paulpcb_default(
```
Comment (zvecr) [inline keyboards/paulpcb/keymaps/default/keymap.c:20]:
```suggestion
    [0] = LAYOUT(
```

File: keyboards/paulpcb/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        KC_ESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_GRAVE, KC_DEL, 
+		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS, KC_MPLY, 
+		KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_ENT, KC_HOME, 
+		KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP, KC_END, 
+		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, KC_B, KC_BSPC, MO(1), KC_RALT, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT_paulpcb_default(
```
Comment (zvecr) [inline keyboards/paulpcb/keymaps/default/keymap.c:27]:
```suggestion
    [1] = LAYOUT(
```

File: keyboards/paulpcb/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_paulpcb_default(
```
Comment (zvecr) [inline keyboards/paulpcb/keymaps/via/keymap.c:20]:
```suggestion
    [0] = LAYOUT(
```

File: keyboards/paulpcb/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        KC_ESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSLS, KC_GRAVE, 
+		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSPC, KC_MPLY, 
+		KC_LCTL, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_ENT, KC_HOME, 
+		KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP, KC_END, 
+		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, KC_B, KC_SPC, MO(1), KC_RGUI, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT_paulpcb_default(
```
Comment (zvecr) [inline keyboards/paulpcb/keymaps/via/keymap.c:27]:
```suggestion
    [1] = LAYOUT(
```

File: keyboards/paulpcb/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        KC_TRNS, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_F13, KC_F14, 
+		KC_CAPS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_DEL, RESET, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_PGUP, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_PGDOWN, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_MEDIA_PREV_TRACK, KC_TRNS, KC_MEDIA_NEXT_TRACK
+    ),
+    [2] = LAYOUT_paulpcb_default(
```
Comment (zvecr) [inline keyboards/paulpcb/keymaps/via/keymap.c:34]:
```suggestion
    [2] = LAYOUT(
```

File: keyboards/paulpcb/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [3] = LAYOUT_paulpcb_default(
```
Comment (zvecr) [inline keyboards/paulpcb/keymaps/via/keymap.c:41]:
```suggestion
    [3] = LAYOUT(
```

File: keyboards/paulpcb/keymaps/via/rules.mk [hunk 8]
```diff
@@ -0,0 +1,9 @@
+VIA_ENABLE = yes
+LTO_ENABLE = yes
+MOUSEKEY_ENABLE = no
+LEADER_ENABLE = no
+FAUXCLICKY_ENABLE = no
+MIDI_ENABLE = no
+BLUETOOTH_ENABLE = no
+KEY_LOCK_ENABLE = no
+TERMINAL_ENABLE = no
```
Comment (zvecr) [inline keyboards/paulpcb/keymaps/via/rules.mk:9]:
```suggestion
```

File: keyboards/paulpcb/paulpcb.h [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+/* This is a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT_paulpcb_default( \
```
Comment (zvecr) [inline keyboards/paulpcb/paulpcb.h:29]:
```suggestion
#define LAYOUT( \
```

File: keyboards/paulpcb/paulpcb.h [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+	K400, K401, K402,       K404,                   K406,       K408, K409, K410,            K411, K412, K414 \
+) { \
+    { K000,  K001,  K002,  K003,  K004,  K005,  K006,  K007,  K008,  K009,  K010,  K011,  K012,  K013,  K014 }, \
+	{ K100,  K101,  K102,  K103,  K104,  K105,  K106,  K107,  K108,  K109,  K110,  K111,  K112,  K113,  K114 }, \
+	{ K200,  K201,  K202,  K203,  K204,  K205,  K206,  K207,  K208,  K209,  K210,  K211,  K212,  KC_NO, K214 }, \
+	{ K300,  K301,  K302,  K303,  K304,  K305,  K306,  K307,  K308,  K309,  K310,  K311,  K312,  KC_NO, K314 }, \
+	{ K400,  K401,  K402,  KC_NO, K404,  K405,  K406,  KC_NO, K408,  K409,  K410,  K411,  K412,  KC_NO, K414 }  \
+}
+
+#define LAYOUT LAYOUT_paulpcb_default
```
Comment (zvecr) [inline keyboards/paulpcb/paulpcb.h:43]:
```suggestion

```

**Review by zvecr — CHANGES_REQUESTED** [review]
`keyboards/paulpcb/.vs/` folder should also be removed.

**Review by AlfieLockrey — COMMENTED** [review]
Thank you for your review, I am pretty new to this side of GitHub so please let me know if I have followed on correctly.

File: keyboards/paulpcb/config.h [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x1311 // AWL ("1311"1999)
```
Comment (drashna) [inline keyboards/paulpcb/config.h:23]: This needs to be changed, as it is already in use: 
https://www.the-sz.com/products/usbid/index.php?v=0x1311&p=&n=

File: keyboards/paulpcb/rules.mk [hunk 12]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/paulpcb/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/paulpcb/config.h (hunk 11 above)
Comment (AlfieLockrey) [reply keyboards/paulpcb/config.h]: Thanks for that, changes have been made.

**Review by drashna — APPROVED** [review]

File: keyboards/paulpcb/readme.md [hunk 13]
```diff
@@ -0,0 +1,27 @@
+# paulpcb
+
+![paulpcb](https://i.imgur.com/BZneAHS.jpeg)
+![AWL Paul keyboard](https://i.imgur.com/dRibVfa.jpeg)
```
Comment (fauxpark) [inline keyboards/paulpcb/readme.md:4]:
```suggestion
![paulpcb](https://i.imgur.com/BZneAHSh.jpeg)
![AWL Paul keyboard](https://i.imgur.com/dRibVfah.jpeg)
```

File: keyboards/paulpcb/readme.md [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+
+The AWL Paul is a reverse Alice style keyboard with an encoder and two 'macro' keys on the RHS. It is designed around simplistic case design that can be 3D printed.
+
+* Keyboard Maintainer: [Alfie_Lockrey](https://github.com/AlfieLockrey)
+* Hardware Supported: PaulPCB V1.0 
+* Hardware Availability: Message me on Discord @ Alfie#6783.**
```
Comment (fauxpark) [inline keyboards/paulpcb/readme.md:10]:
```suggestion
* Hardware Availability: Message me on Discord @ Alfie#6783.
```

File: keyboards/paulpcb/info.json [hunk 15]
```diff
@@ -0,0 +1,84 @@
+{
+    "keyboard_name": "AWL PaulPCB",
+    "url": "https://github.com/AlfieLockrey",
+    "maintainer": "AlfieLockrey",
+    "width": 18.5,
+    "height": 6.6,
+    "layouts": {
+        "LAYOUT": {
+            "key_count": 70,
```
Comment (fauxpark) [inline keyboards/paulpcb/info.json:9]:
```suggestion
    "layouts": {
        "LAYOUT": {
```

File: keyboards/paulpcb/rules.mk [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration 
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/paulpcb/rules.mk:16]:
```suggestion
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/paulpcb/readme.md (hunk 13 above)
Comment (AlfieLockrey) [reply keyboards/paulpcb/readme.md:4]: I can see that this changes the resolution of the image, is this standard practice or written anywhere or personal preference? I couldn't see it in the PR Checklists.

File: keyboards/paulpcb/readme.md (hunk 14 above)
Comment (AlfieLockrey) [reply keyboards/paulpcb/readme.md]: Altered this to a single line, including the line below in a separate commit

**Review by fauxpark — APPROVED** [review]

File: keyboards/paulpcb/readme.md (hunk 13 above)
Comment (fauxpark) [reply keyboards/paulpcb/readme.md]: In general you should avoid making people download massive images.
image

File: keyboards/paulpcb/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+};
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    } else {
+      tap_code(KC_VOLD);
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/paulpcb/keymaps/default/keymap.c:43]:
```suggestion
```

File: keyboards/paulpcb/paulpcb.c [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "paulpcb.h"
```
Comment (drashna) [inline keyboards/paulpcb/paulpcb.c:17]:
```suggestion
#include "paulpcb.h"

bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (clockwise) {
      tap_code(KC_VOLU);
    } else {
      tap_code(KC_VOLD);
    }
    return true;
}
```

File: keyboards/paulpcb/keymaps/via/keymap.c [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+};
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    } else {
+      tap_code(KC_VOLD);
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/paulpcb/keymaps/via/keymap.c:58]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/paulpcb/rules.mk [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration 
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/paulpcb/rules.mk:21]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/paulpcb/info.json [hunk 21]
```diff
@@ -0,0 +1,81 @@
+{
```
Comment (noroadsleft) [inline keyboards/paulpcb/info.json:1]: If this firmware flashes as intended to your PCB as far as key assignments, then this file is wrong.

image

The JSON objects need to be in the same order that `keymap.c` uses.

File: keyboards/paulpcb/rules.mk [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration 
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes		# Encoder enable
+
```
Comment (noroadsleft) [inline keyboards/paulpcb/rules.mk:23]:
```suggestion
ENCODER_ENABLE = yes        # Encoder enable
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/paulpcb/rules.mk [hunk 23]
```diff
@@ -0,0 +1,23 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration 
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/paulpcb/rules.mk:18]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**